62_CAD/DrawResourcesFiller.cpp

```diff
@@ -455,22 +455,31 @@ void DrawResourcesFiller::finalizeTextureCopies(SIntendedSubmitInfo& intendedNex
 	for (uint32_t i = 0; i < textureCopies.size(); i++)
 	{
 		auto& textureCopy = textureCopies[i];
-		asset::IImage::SBufferCopy region = {};
-		region.imageSubresource.aspectMask = asset::IImage::EAF_COLOR_BIT;
-		region.imageSubresource.mipLevel = 0u;
-		region.imageSubresource.baseArrayLayer = textureCopy.index;
-		region.imageSubresource.layerCount = 1u;
-		region.bufferOffset = 0u;
-		region.bufferRowLength = textureCopy.imageExtent.x;
-		region.bufferImageHeight = 0u;
-		region.imageExtent = { textureCopy.imageExtent.x, textureCopy.imageExtent.y, textureCopy.imageExtent.z };
-		region.imageOffset = { 0u, 0u, 0u };
-
-		m_utilities->updateImageViaStagingBuffer(
-			intendedNextSubmit, 
-			textureCopy.srcBuffer->getPointer(), nbl::ext::TextRendering::TextRenderer::MSDFTextureFormat,
-			msdfImage.get(), IImage::LAYOUT::TRANSFER_DST_OPTIMAL, 
-			{ &region, &region + 1 });
+		for (uint32_t mip = 0; mip < textureCopy.image->getCreationParameters().mipLevels; mip++)
```

Comment:
I think you can just get the regions from ICPUImage instead of calculating manually